Scripts/SL-Helpers.lua

```diff
@@ -572,3 +572,83 @@ end
 IsAutoplay = function(player)
 	return GAMESTATE:GetPlayerState(player):GetPlayerController() ~= "PlayerController_Human"
 end
+
+
+-- -----------------------------------------------------------------------
+CalculateExScore = function(player)
+	local pn = ToEnumShortString(player)
+	local stats = STATSMAN:GetCurStageStats():GetPlayerStageStats(pn)
+
+	local total_points = 0
+	local total_possible = 0
+
+	local TNS = { "W1", "W2", "W3", "W4", "W5", "Miss" }
+
+	-- First get tap notes.
+	for index, window in ipairs(TNS) do
+		local number = stats:GetTapNoteScores( "TapNoteScore_"..window )
+
+		-- The W0 window needs to be emulated in ITG mode.
+		-- We already track note offsets so use that to compute the W0 count.
+		-- TODO(teejusb): Explicitly track this value instead of computing the count here
+		-- as it will be useful to display the count during gameplay.
+		if window == "W1" and SL.Global.GameMode == "ITG" then
+			local offsets = SL[ToEnumShortString(player)].Stages.Stats[SL.Global.Stages.PlayedThisGame + 1].sequential_offsets
+
+			local W0_count = 0
+
+			for time_offset in ivalues(offsets) do
+				local offset = time_offset[2]
+
+				local W0 = SL.Preferences["FA+"]["TimingWindowSecondsW1"] + SL.Preferences["FA+"]["TimingWindowAdd"]
```

Comment:
I think this should take take TimingWindowScale into account. Basically what GetTimingWindow is doing:
```
prefs["TimingWindowSecondsW"..n] * scale + prefs.TimingWindowAdd
```

Reply:
We only need to do this for the W0 window in ITG mode (since this calculation is already implicitly done in the judgment counts) so I don't need a whole function for it but added.